src/com/visualization/barchart/bar.js

```diff
@@ -0,0 +1,42 @@
+import {h, Component} 				from 'preact/preact';
+
+export default class Bar extends Component {
+
+	constructor( props ) {
+		super(props);
+
+    }
+
+    render( props ) {
+
+        // it's valid for height/width/index to be zero so we have to check it against undefined.
+        if ( !(
+			props &&
+			props['height']	!= undefined &&
+			props['index']	!= undefined &&
+			props['width']	!= undefined &&
+			props['color']
+		) ) {
```

Comment:
Prefer
```js
if ( !(props && (props.height != null) && (props.index != null) && (props.width != null) && (props.color != null)) ) {
```
or
```js
if ( !(
    props && 
    (props.height != null) &&
    (props.index != null) && 
    (props.width != null) && 
    (props.color != null)
) ) {
```
Noting that:
* The `!=` operator actually checks for both `undefined` and `null`, and `null` is preferred in this case.
* I'm guessing the `props.color` check also needs to be a `!= null` test

Alternatively, if a non-zero/non-empty-string is also a worthwhile thing to be testing for, consider:
```js
if ( !(props && props.height && props.index && props.width && props.color) ) {
```
Though without knowing the code, I could be wrong and `props.index` may be something where `0` should be legal.

Reply:
0 is legal for index height and width but not for color.
I keep forgetting that you don't need to use the `props.['key']` syntax